pkg/fqdn/cache.go

```diff
@@ -829,6 +833,7 @@ func (zombies *DNSZombieMappings) Upsert(expiryTime time.Time, addr netip.Addr,
 		zombie = &DNSZombieMapping{
 			Names:           ciliumslices.Unique(qname),
 			IP:              addr,
+			AliveAt:         zombies.lastCTGCUpdate,
 			DeletePendingAt: expiryTime,
```

Comment:
side note, the delete pending is exposed to users via `cilium-dbg fqdn cache list` and may give the impression that the expiry time is a long time in the future for high-TTL entries, but of course when in an overlimit scenario Cilium may choose not to observe this delete pending timestamp... and just GC the entry anyway. I think this is the right behavior given the scenario, even if it might give slightly misleading data to an observant user.